src/Library/Lod/LodReader.h

```diff
@@ -21,20 +18,34 @@ class LodReader final {
  public:
     static std::unique_ptr<LodReader> open(const std::string &filename);
 
-    inline ~LodReader() {
-        if (nullptr != _fp) {
-            fclose(_fp);
-        }
-    }
+    inline ~LodReader() {}
 
     bool exists(const std::string &filename) const;
     Blob read(const std::string &filename);
 
+    void ls() const;
```

Comment:
Orthogonality striketh back.

This is a utility class, and it should be up to the caller what to do with the results of an `ls` call.

If you want a debug-only function, rename this one to something like `void debugDump(FILE *) const`.

If you want a function that lists the files, then use `std::vector<std::string> ls() const`.